libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -86,7 +93,57 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     });
   }
 
+  private getContentMaxHeight(): number {
+    const ionModalWrapper = this.elementRef.nativeElement.parentElement;
+    const ionModal = ionModalWrapper.parentElement;
+    const ionModalPaddingTop = this.windowRef
+      .getComputedStyle(ionModal)
+      .getPropertyValue('padding-top');
+
+    let availableSpace =
+      document.documentElement.clientHeight -
+      parseInt(ionModalPaddingTop) -
+      (this.ionContentElement.nativeElement.getBoundingClientRect().top -
+        this.elementRef.nativeElement.getBoundingClientRect().top);
+
+    if (this.embeddedFooterElement) {
+      availableSpace -= this.embeddedFooterElement.getBoundingClientRect().height;
+    }
+    return availableSpace;
+  }
+
+  private async setHeightOfContent(contentHeight: number) {
+    const modalElementRef = this.elementRef.nativeElement.parentElement;
+    const contentMaxHeight = this.getContentMaxHeight();
+
+    if (contentHeight < contentMaxHeight && modalElementRef) {
+      this.renderer.removeClass(modalElementRef, 'content-exceeds-viewport');
+    } else if (modalElementRef) {
+      this.renderer.addClass(modalElementRef, 'content-exceeds-viewport');
+    }
+
+    const scrollElement = await this.ionContent.getScrollElement();
+    const scrollElementStyles = this.windowRef.getComputedStyle(scrollElement);
+
+    const scrollElementPaddingTopAndBottom =
+      parseInt(scrollElementStyles.getPropertyValue('padding-top')) +
+      parseInt(scrollElementStyles.getPropertyValue('padding-bottom'));
+
+    const minHeight = Math.min(contentHeight + scrollElementPaddingTopAndBottom, contentMaxHeight);
+    this.renderer.setStyle(this.ionContentElement.nativeElement, 'min-height', `${minHeight}px`);
+  }
+
+  private onEmbededComponentElementResize() {
+    this.resizeObserverService.observe(this.embeddedComponentElement, (entry) => {
```

Comment:
Is resizeObserverService resilliant to null values? (The method signature doesn't hint at that...) Or do you need to add a null check for `this.embeddedComponentElement` ?